kernel/Kconfig

```diff
@@ -331,6 +331,7 @@ config BOOT_BANNER
 
 config BOOT_DELAY
 	int "Boot delay in milliseconds"
+	depends on MULTITHREADING
```

Comment:
As I can see it doesn't actually depend on MULTITHREADING. Probably it's worth to drop this restriction?

Reply:
My understanding is that this delay is so that other threads can run and perform setup operations. Without `MULTITHREADING` what reason is there for the delay?

Reply:
It makes sense for certain tests. Please leave it without any dependency on multithreading.

Reply:
But the previous implementation did not delay at all when MULTITHREADING was disabled. Dropping that dependency would be a change in behaviour. This PR is only meant to cleanup the implementation.

Reply:
Not sure why we have that limitation. I'd say a clean-up could also simplify this by removing that condition.

Reply:
I am with @JordanYates, do not change behaviour in a cleanup PR. This keeps biting us.. If there is something to improve or fix, please do that in a different PR. In this case, I remember the issue and it was added for a reason: https://github.com/zephyrproject-rtos/zephyr/issues/14454

Reply:
Ok, fair enough :)